vadl/test/vadl/ast/TypecheckerProcessorTest.java

```diff
@@ -41,13 +41,9 @@ void shouldThrow_duplicatedDefinitions() {
         firmware = {}
         """;
     var spec = base.formatted(body);
-    var ast = Assertions.assertDoesNotThrow(() -> VadlParser.parse(spec), "Cannot parse input");
-    var typechecker = new TypeChecker();
-    var throwable = assertThrows(DiagnosticList.class, () -> typechecker.verify(ast));
-    TestUtils.assertErrors(throwable,
-        "Contains multiple `firmware` definitions",
-        "Contains multiple `start` definitions"
+    var throwable = Assertions.assertThrows(DiagnosticList.class, () -> VadlParser.parse(spec)
     );
+    TestUtils.assertErrors(throwable, "Symbol name already used: start");
```

Comment:
@benjaminkasper99 The problem here is that this issue is now already found during symbol resolution instead of the typechecking.

I don't know if you still want to leave this test in but because the old symbol collector skipped some parts of the trees it previously was left as an task for the typechecker.